### ISSUE TITLE
full on-boarding/ register a name use case

### ISSUE THREAD

**Issue by guylepage3** [blockstack/blockstack-browser#557]
## Register a name process

After I've connected my dropbox i am left stagnant on the storage screen.

The following does not include the **"installation process"**

Process should guide user through name creation
1. **User goes through all of the on-boarding screens**
    1. **Welcome screen** — Click "Get Started" button
    2. **Create keychain screen** — Click "Create Keychain" button
    3. **Choose a password screen** — Enter valid password and click "Continue" button
    4. **Enter email address** — Enter valid email address and click "Yes" button
    5. 🚩 **Home screen** — _I'm confused. What next? Why did I land here? Why do i need to create a profile?_ — There should be another set of screens or a glaring notification that tells me what to do next and walk me through it.
2. **Create a username**
    1. **Go to Profiles App** — _I don't know that this app has any relation to the "Browser."_ There should be some sort of inclusion between the browser and it's apps or I won't know that they are a "universal" app that is native to the Browser. Right now it looks like it's a different company's app.
    2. **Create profile** — Click the "+ Create" button to create a profile.
    3. 🚩 **_Great i'm done my user was created_**. I don't need to do anything else. I can view and edit my profile — If I am done creating a profile I am good and therefore don't need to do anything else.
    4. 🚩 **Add username** — Alert says "Please go to the Storage app and connect a storage provider." — There should be a button or modal window guiding me through this process. I have no idea that the "alert" banner is clickable. If I did not know that this was clickable I'd be lost. — Click the "Please go to the Storage app and connect a storage provider." link. Maybe add a 
    5. 🚩 **Storage app screen** — _Why am I here? I need guidance. What do I do again? I forgot why I landed here._ — Click the "Connect to Dropbox." 🚩 _Oh wait.. Why is this important? I don't want to connect to Dropbox._
    6. 🚩 **Dropbox API screen** — Click the Dropbox account button that I'd like. _Wait.. Why was I pushed to dropbox.com?? Did I complete my profile? What's going on?_
    7. 🚩 **Browser home screen** — _Cool. What happened? Did I complete something? How do I know what happened? What next?_
    8. 🚩 **Profiles list view** — Click "Add username" button — _Why? I don't want to._
    9. 🚩 **Add username screen** — Alert says "You need to deposit at least 0.01 bitcoins before you can register a username. Click here to go to your wallet or send bitcoins directly to null." _What? Why? Wasn't I here already?_ — Again, there should be a button or modal window guiding me through this process. I have no idea that the "alert" banner is clickable. If I did not know that this was clickable I'd be lost.
    10. **Wallet app page** — _How did I end up here in a different app and interface. Am I still in the Blockstack browser? What do I do next? Am I supposed to copy and paste the address somewhere?_ — Copied the bitcoin address.
    11. **Coinbase home page** — Navigated to coinbase.com
    12. **Coinbase sign in page** — Logged in to coinbase.com
    13. **Coinbase dashboard** — Clicked "Send & Receive" button
    14. **Coinbase send & receive screen** — Pasted in the bitcoin address of the blockstack wallet
    15. **Blockstack browser wallet** — Forgot the minimum amount I was supposed to send to the Blockstack wallet. Went back to the Blockstack Browser to view the minimum amount I was supposed to send to the Blockstack wallet
    16. **Coinbase send & receive screen** — Entered the minimum amount to be sent to the Blockstack wallet
    17. **Coinbase send & receive screen** — Entered transaction note
    18. **Coinbase send & receive screen** — Clicked "Send" & received a successfully sent confirmation message from coinbase — _Now what? I am assuming I am to go back to the Blockstack Browser to view my wallet._
    19. **Blockstack browser wallet** — Went back to the Blockstack Browser and I don't see any change. — _Why don't I see a change in my wallet? Where did my money go? When do the funds arrive in my Blockstack wallet?_
    20. 🚩🚩 **Major flag** 🚩🚩 — _What's happening? I've been refreshing my browser/blockstack browser for 5 minutes now.. Where are my funds? My wallet is showing 0 BTC. Please help._
    21. 🚩 **Home screen** — If the user has to wait for their wallet to fill, there is a good chance they will either quit the process all together or come back to the browser and land on a home screen with no indicator that their wallet has received funds — We may want to add an app icon notification in this instance to let the user know that they have funds.
    22. **Profiles list view** — There probably should be a visual mechanism to let the user know that they have now completed the steps to being able to "Add username" to their profile(s) — Click "Add username" link.
    23. **Add username screen** — I have completed all steps and I am now able to register a username.
    24. **Username is available** — Alert box indicates that "[username].id costs ~0.007 btc to register." We may want to reword this as I have no idea if that means I am able to register or not. Also, I would like a warning that I will be spending BTC here and there are no ways to receive a refund. All sales are final with a link to disclaimer of why they are final. i.e. your BTC are being burned.
    25. **Submitting registration to Core node** — _Why do I care about this? I just want a name._
    26. **Congratulations screen/alert** — At the end of the process I have received an alert box indicating the following "Congrats! Your name is preordered! Registration will automatically complete over the next few hours." — _[sad face] I am bummed... I was hoping to get my name right now_ — Should we rethink the wording. "Congrats!" you have to wait seems a tiny bit misleading. I am also bummed that the wait window is over the next few hours.. I was really anticipating to use my shiny new username right after I paid so much for it.
    27. **Profiles list view** — I now see that my new .id's username is pending.. — _I would like to see a countdown indicator of some kind. What's the status? How much longer do I have to wait?_

**Tue, Jun 27 — 11:20am**
- I am waiting for my new id to go through.

**Questions?**
- I love my new blockstack id. How do i save my keys for this new id? I would like to see an indicator when I click the "Edit" button.

**Comment by guylepage3** [comment]
Paused while waiting for funds to arrive in my blockstack browser wallet.

**Comment by guylepage3** [comment]
continuing on from yesterday's name registration attempt.

**Comment by guylepage3** [comment]
Stuck waiting or my name to go through..

Here's my progress for registering a name today.

21. 🚩 **Home screen** — If the user has to wait for their wallet to fill, there is a good chance they will either quit the process all together or come back to the browser and land on a home screen with no indicator that their wallet has received funds — We may want to add an app icon notification in this instance to let the user know that they have funds.
    22. **Profiles list view** — There probably should be a visual mechanism to let the user know that they have now completed the steps to being able to "Add username" to their profile(s) — Click "Add username" link.
    23. **Add username screen** — I have completed all steps and I am now able to register a username.
    24. **Username is available** — Alert box indicates that "[username].id costs ~0.007 btc to register." We may want to reword this as I have no idea if that means I am able to register or not. Also, I would like a warning that I will be spending BTC here and there are no ways to receive a refund. All sales are final with a link to disclaimer of why they are final. i.e. your BTC are being burned.
    25. **Submitting registration to Core node** — _Why do I care about this? I just want a name._
    26. **Congratulations screen/alert** — At the end of the process I have received an alert box indicating the following "Congrats! Your name is preordered! Registration will automatically complete over the next few hours." — _[sad face] I am bummed... I was hoping to get my name right now_ — Should we rethink the wording. "Congrats!" you have to wait seems a tiny bit misleading. I am also bummed that the wait window is over the next few hours.. I was really anticipating to use my shiny new username right after I paid so much for it.
    27. **Profiles list view** — I now see that my new .id's username is pending.. — _I would like to see a countdown indicator of some kind. What's the status? How much longer do I have to wait?_

**Tue, Jun 27 — 11:20am**
- I am waiting for my new id to go through.

**Questions?**
- I love my new blockstack id. How do i save my keys for this new id? I would like to see an indicator when I click the "Edit" button.

**Comment by guylepage3** [comment]
**Tue, Jun 27 — 1:33pm**
Just checked to see if I've received my fancy new username.. No dice.

**Comment by guylepage3** [comment]
Wed, Jun 28, 2017 — 2pm
Apparently my id went through but I did not receive a notification and now I am unable to view my profile. See: https://github.com/blockstack/blockstack-browser/issues/575

**Comment by guylepage3** [comment]
**Thu, Jun 29, 2017 — 2:40pm**
Still not able to view my profile. Waiting for @larrysalibra to have time to get around to my issue. See: https://github.com/blockstack/blockstack-browser/issues/575

**Comment by pstan26** [comment]
Add screenshots if you want this to be carefully read by more people. Its good description, just need accompanying visuals IMO. Wouldn't spend more than 20 min on it though.

**Comment by pstan26** [comment]
Here is how Airbnb mapped user journeys. Possibly excessive but interesting: https://www.fastcompany.com/3002813/how-snow-white-helped-airbnbs-mobile-mission

**Comment by guylepage3** [comment]
>Add screenshots if you want this to be carefully read by more people. Its good description, just need accompanying visuals IMO.

@pstan26 very good point. would take 30 mins or so.

@shea256 @larrysalibra do you feel there's value in adding the screen shots?

**Comment by larrysalibra** [comment]
I don’t think you need to add screenshots @guylepage3 - I’ve read through it and I think it’s pretty clear we have a lot of steps and many points of user confusion. Thanks for compling this!

**Comment by guylepage3** [comment]
Ok. So my profile is fully viewable. Looks as though my name registration process took around 7 days. But the great news is, that when I go to register another name, more than likely it will be a smoother process by a few days at least. :)
